src/vos/ilog.c

```diff
@@ -1049,7 +991,12 @@ ilog_modify(daos_handle_t loh, const struct ilog_id *id_in,
 
 	root = lctx->ic_root;
 
-	if (root->lr_tree.it_embedded) {
+	D_DEBUG(DB_IO, "%s in incarnation log: epoch:" DF_U64
+		" tree_version: %d\n", opc_str[opc], id_in->id_epoch,
+		ilog_mag2ver(root->lr_magic));
+
+	if (root->lr_tree.it_embedded && root->lr_id.id_epoch <= epr->epr_hi
+	    && root->lr_id.id_epoch >= epr->epr_lo) {
```

Comment:
this code block can be moved into the "else" at L1026, because the @visibility is only used by that code block.